great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -47,77 +41,52 @@ def __init__(
         )
 
         self._runtime_keys = runtime_keys
+        self._refresh_data_references_cache()
 
     def _refresh_data_references_cache(self):
-        """"""
-        # Map data_references to batch_definitions
-        data_reference: str = self._get_data_reference_list()[0]
-        mapped_batch_definition_list: List[
-            BatchDefinition
-        ] = self._map_data_reference_to_batch_definition_list(
-            data_reference=data_reference, data_asset_name=None
-        )
-        self._data_references_cache = {data_reference: mapped_batch_definition_list}
+        self._data_references_cache = {}
 
     def _get_data_reference_list(
         self, data_asset_name: Optional[str] = None
     ) -> List[str]:
-        """List objects in the underlying data store to create a list of data_references.
-
-        This method is used to refresh the cache.
-        """
-        return [""]
+        if data_asset_name:
+            return self._get_data_reference_list_from_cache_by_data_asset_name(
+                data_asset_name
+            )
+        else:
+            data_reference_list = []
+            for data_asset_name in self.get_available_data_asset_names():
+                data_reference_list += (
+                    self._get_data_reference_list_from_cache_by_data_asset_name(
+                        data_asset_name
+                    )
+                )
+            return data_reference_list
 
     def _get_data_reference_list_from_cache_by_data_asset_name(
         self, data_asset_name: str
     ) -> List[str]:
         """Fetch data_references corresponding to data_asset_name from the cache."""
-        batch_definition_list: List[
-            BatchDefinition
-        ] = self._get_batch_definition_list_from_batch_request(
-            batch_request=BatchRequestBase(
-                datasource_name=self.datasource_name,
-                data_connector_name=self.name,
-                data_asset_name=data_asset_name,
-            )
+        data_references_for_data_asset_name = self._data_references_cache.get(
+            data_asset_name, None
```

Comment:
@Shinnnyshinshin `None` is already implied by `get()`.

Reply:
🙇🏼